### PR TITLE
Upgrade typescript to 3.7.2

### PR DESCRIPTION
**Pull request by danrosenthal**
[⏫ TypeScript 3.6.3](https://devblogs.microsoft.com/typescript/announcing-typescript-3-6/)

### CODE REVIEW

**Comment by danrosenthal** [comment]
```
#!/bin/bash -eo pipefail
yarn run check:ci
yarn run v1.10.1
$ npm-run-all lint type-check test:ci
$ sewing-kit lint

=============

WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.6.0

YOUR TYPESCRIPT VERSION: 3.6.3

Please only submit bug reports when using the officially supported version.

=============


error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
ERROR: "lint" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Exited with code 1
```

**Comment by danrosenthal** [comment]
I think past me could be responsible for this issue. What do you think @BPScott?

https://github.com/Shopify/polaris-react/pull/1650

**Comment by BPScott** [comment]
Yeah that looks likely, we've still not got a fix for that into eslint-plugin-shopify yet so I we still need that resolution, but we might be able to bump the version that it points at

**Comment by BPScott** [comment]
Yay a new SK with a new eslint-plugin-shopify has been released that removes the need for that resolutions dance. I'll see about updating in the new week or so

**Comment by BPScott** [comment]
https://github.com/Shopify/polaris-react/pull/2369 is the PR for the SK update, but alas we're blocked on that because thanks annoyances with the prettier-vscode version (we've updated to eslint 6 but they haven't and that causes problems)

File: src/components/ContextualSaveBar/ContextualSaveBar.tsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
+import {
+  ContextualSaveBarProps as ContextualSaveBarProps1,
+  useFrame,
+} from '../../utilities/frame';
 
 // The script in the styleguide that generates the Props Explorer data expects
 // a component's props to be found in the Props interface. This silly workaround
 // ensures that the Props Explorer table is generated correctly, instead of
-// crashing if we write `ContextualSaveBar extends React.Component<ContextualSaveBarProps>`
-export interface ContextualSaveBarProps extends ContextualSaveBarProps {}
```
Comment (BPScott) [inline src/components/ContextualSaveBar/ContextualSaveBar.tsx]: TS 3.7 stopped you having a local type with the same name as an import https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#local-and-imported-type-declarations-now-conflict

File: src/components/DropZone/tests/DropZone.test.tsx [hunk 2]
```diff
@@ -469,6 +469,9 @@ function setBoundingClientRect(size: keyof typeof widths) {
         left: 0,
         bottom: 0,
         right: 0,
+        x: 0,
```
Comment (BPScott) [inline src/components/DropZone/tests/DropZone.test.tsx]: Updates to the type that `Element.getBoundingClientRect()` returns added these new values so we need to add them to the mock (this crops up a few times) https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#dom-changes

File: src/components/ThemeProvider/ThemeProvider.tsx [hunk 3]
```diff
@@ -32,10 +32,10 @@ export function ThemeProvider({
     [customProperties, themeConfig, unstableGlobalTheming],
   );
 
-  // We want these values to be `null` instead of `undefined` when not set.
+  // We want these values to be empty string instead of `undefined` when not set.
   // Otherwise, setting a style property to `undefined` does not remove it from the DOM.
-  const backgroundColor = customProperties['--p-surface-background'] || null;
-  const color = customProperties['--p-text-on-surface'] || null;
+  const backgroundColor = customProperties['--p-surface-background'] || '';
```
Comment (BPScott) [inline src/components/ThemeProvider/ThemeProvider.tsx]: backgroundColor is now not nullable. color can be nullable but i figured we might as well make this consistent as it seems to behave the same in chrome for me. @tmlayton  can you confirm the intended behaviour still works for you?

**Comment by BPScott** [comment]
While 3.7.x is super shiny, we might need update to [babel 7.7.x](https://babeljs.io/blog/2019/11/05/7.7.0#typescript-37-10543-https-githubcom-babel-babel-pull-10543-10545-https-githubcom-babel-babel-pull-10545) and poke at our babel config for full support, as the storybook build is done using babel rather than TSC, but that's a story for later

File: src/components/ThemeProvider/ThemeProvider.tsx (hunk 3 above)
Comment (tmlayton) [reply src/components/ThemeProvider/ThemeProvider.tsx]: If the 🎩 looks good I’m fine with it. Try toggling global theming on and off and make sure it works.

File: src/components/ContextualSaveBar/ContextualSaveBar.tsx (hunk 1 above)
Comment (amrocha) [reply src/components/ContextualSaveBar/ContextualSaveBar.tsx]: I thumbs upped your comment but I do not thumbs up anything that's going on here

File: src/components/Toast/Toast.tsx [hunk 4]
```diff
@@ -2,16 +2,16 @@ import React, {useRef} from 'react';
 import {Toast as AppBridgeToast} from '@shopify/app-bridge/actions';
 
 import {DEFAULT_TOAST_DURATION} from '../Frame';
-import {ToastProps, useFrame} from '../../utilities/frame';
+import {ToastProps as ToastProps1, useFrame} from '../../utilities/frame';
```
Comment (amrocha) [inline src/components/Toast/Toast.tsx]: thanks I hate it

**Pull request merged**